userdocs/diagnostics/string-interpolation-conformance.md

```diff
@@ -0,0 +1,10 @@
+# Conforming to `StringInterpolationProtocol`
+Types can implement custom interpolation behavior in string literals by conforming to `StringInterpolationProtocol`. In addition to its formal requirements, `init(literalCapacity:interpolationCount:)` and `appendLiteral(_:)`, `StringInterpolationProtocol` has an additional, informal requirement, `appendInterpolation`. String interpolations using `\()` syntax are translated into calls to matching `appendInterpolation` methods.
```

Comment:
The beginning of this sentence makes it sound like all you need to do to get interpolation to work is conform to `StringInterpolationProtocol`. I think I would expand on this slightly to give a little more background information:

> A type conforming to `ExpressibleByStringInterpolation` uses a helper type called `StringInterpolation` perform its interpolation. Many types can use `DefaultStringInterpolation`, which implements `String`'s interpolation behavior; to provide custom behavior instead, provide a custom type conforming to `StringInterpolationProtocol`.
>
> In addition to its formal requirements…